R/tbl_split.R

```diff
@@ -1,66 +1,236 @@
-#' Split gtsummary table
+#' Split gtsummary table bz rows and/or columns
 #'
 #' `r lifecycle::badge('experimental')`\cr
-#' The `tbl_split` function splits a single gtsummary table into multiple tables.
+#' The `tbl_split_by_rows` and `tbl_split_by_columns`` functions split a single
+#' gtsummary table into multiple tables.
+#' Both horizontal (column-wise, (that is, splits by columns in
+#' `x$table_body`) and vertical (row-wise) splits are possible.
 #' Updates to the print method are expected.
 #'
-#' @param x (`gtsummary`)\cr
-#'   gtsummary table
+#' @param x (`gtsummary` or `list`)\cr
+#'   gtsummary table.
 #' @param variables ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
 #'   variables at which to split the gtsummary table rows (tables
-#'   will be separated after each of these variables)
-#' @inheritParams rlang::args_dots_empty
+#'   will be separated after each of these variables).
+#' @param keys ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   columns to be repeated in each table split. It defaults to `first_column(x)`
+#'   if missing (usually label column).
+#' @param groups (list of `character` vectors)\cr
+#'   list of column names that appear in `x$table_body`.
+#'   Each group of column names represent a different table in the output list.
+#' @param footnotes (`string`)\cr
+#'   can be either `"first"`, `"each"`, or `"last"`, to locate global footnotes
+#'   only on the first, in each, or in the last table, respectively. It defaults
+#'   to `"each"`. Reference footnotes are always present wherever they appear.
+#' @param caption (`string`)\cr
+#'   similarly to footnotes, makes the titles, and subtitles appear only on the
+#'   `"first"` or `"each"` table. It defaults to `"each"`.
 #'
-#' @return `tbl_split` object
-#' @name tbl_split
+#' @return `tbl_split` object. They can also be identified further by subclass
+#'   definitions `tbl_split_by_rows` and `tbl_split_by_columns`.
+#'
+#' @details
+#' Run `gtsummary::show_header_names()` to print all column names to split by.
 #'
 #' @examples
-#' tbl <-
-#'   tbl_summary(trial) |>
-#'   tbl_split(variables = c(marker, grade))
+#' # create standard table
+#' tbl <- tbl_summary(trial, by = trt)
+#'
+#' # split by rows
+#' tbl_list_rows <- tbl |>
+#'   tbl_split_by_rows(variables = c(marker, grade))
+#'
+#' # print column names
+#' gtsummary::show_header_names(tbl)
+#'
+#' # split by columns
+#' tbl_list_cols <- tbl |>
+#'   tbl_split_by_columns(groups = list("stat_1", "stat_2"))
+#'
+#' # take the list of splitted rows and split again by columns
+#' tbl_list_rows_cols <- tbl_list_rows |>
+#'   tbl_split_by_columns(groups = list("stat_1", "stat_2"))
+#'
+#' # Handle footnotes
+#'
+#' @name tbl_split_by
 NULL
 
 #' @export
-#' @rdname tbl_split
-tbl_split <- function(x, ...) {
-  UseMethod("tbl_split")
+#' @rdname tbl_split_by
```

Comment:
It was conflicting with tbl_split. Is it ok?

Reply:
Aiyah, of course. yeah that is good.